lib/tasks/gitlab/check.rake

```diff
@@ -276,6 +277,45 @@ namespace :gitlab do
         fix_and_rerun
       end
     end
+    
+    def check_upload_perms
+      print "Upload directory permissions correct? ... "
+
+      upload_path = File.realpath(Rails.root.join('public/uploads'))
+      upload_path_tmp = File.join(upload_path, 'tmp')
+
+      if File.stat(upload_path).mode == 040750
+        unless Dir.exists?(upload_path_tmp)
+          puts 'skipped (no tmp uploads folder yet)'.magenta
+          return
+        end
+
+        # if tmp upload dir has incorrect permissions, assume others do as well
+        if File.stat(upload_path_tmp).mode == 040755 && File.owned?(upload_path_tmp) # verify drwxr-xr-x permissions
+          puts "yes".green
+        else
+          puts "no".red
+          try_fixing_it(
+            "sudo chown -R #{gitlab_user} #{upload_path}",
```

Comment:
Does the group name also need to be specified (e.g. `gitlab-www`)? See: https://gitlab.com/gitlab-org/omnibus-gitlab/issues/254